lib/msal-browser/src/cache/DatabaseStorage.ts

```diff
@@ -60,7 +69,7 @@ export class DatabaseStorage<T>{
      * Retrieves item from IndexedDB instance.
      * @param key 
      */
-    async get(key: string): Promise<T> {
+    async get<T>(tableName: DBTableNames, key: string): Promise<T> {
```

Comment:
If table name is passed in constructor, why does it need to be passed here?

Reply:
I would say each databasestorage instance should point to a specific table

Reply:
The constructor receives an array of (supported) table names to create all the tables MSAL has. The API's receive a single (supported) table name so they know which table to perform the operation on.

Reply:
>I would say each databasestorage instance should point to a specific table

@jasonnutter what do you think the benefit there would be? As it stands now, CryptoOps has a single DB object that it treats as "cache". I think it's easier to keep track of a single DB with different tables than to have multiple Databases with a single table each.

Reply:
Ah right, misread the code and forgot we'll have a db with multiple tables, as opposed to multiple dbs.

Reply:
Take the Bound RT use as an example to illustrate:

```js
this.cache = new DatabaseStorage(dbName, [DBTableNames.asymmetricKeys, DBTableNames.symmetricKeys], dbVersion);
const stk_jwk = this.cache.get<CryptoKeyPair>(DBTableNames.asymmetricKeys, stkKid);
const sessionKey = this.cache.get<CryptoKey>(DBTableNames.symmetricKeys, sessionKeyKid);

// Clearing IndexedDB:

this.cache.clear();
```

rather than

```typescript
export type KeyStore = {
   asymmetricKeys: DatabaseStorage<CachedKeyPair>;
   symmetricKeys: DatabaseStorage<CryptoKey>;
}

this.cache: KeyStore = {
    asymmetricKeys: new DatabaseStorage<CachedKeyPair>(dbName, DBTableNames.asymmetricKeys, ...),
    symmetricKeys: new DatabaseStorage<CryptoKey>(dbName, DBTableNames.symmetricKeys, ...)
};

const stk_jwk = this.cache.asymmetricKeys.get(stkKid);
const sessionKey = this.cache.symmetricKeys.get(sessionKeyKid);

// Clearing IndexedDB
Object.keys(this.cache).forEach((tableName: string) => this.cache[tableName].clear());
```

With the former, we just update the DatabaseStorage APIs and remove the type restriction on the class. To add support for new tables, we just add them to the enum.

With the latter we need to update the CryptoOps constructor every time to add a new DatabaseStorage instance, have a type (i.e. KeyStore) to keep track of the DbStorage objects and then we need to iterate over these objects to clear each table.


I think these approaches are functionally equivalent, but I think the former is closer to what IndexedDB's intended usage is.

Reply:
I personally like different `DataBaseStorage` instances for each table. Especially since we know we will be adding main cache support for IDB in the future.

Reply:
> I think these approaches are functionally equivalent, but I think the former is closer to what IndexedDB's intended usage is.

True, but we should build the abstractions we want, not necessarily match what we've been given.

Reply:
Switched implementation to have `DatabaseStorage` instances represent a single table each, all still inside a single IndexedDB database.